clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -176,20 +176,32 @@ bool
 check_branch_target_after_branch()
 {
     std::cerr << "Testing branch targets\n";
+    constexpr memref_tid_t TID = 1;
+    constexpr uintptr_t TIMESTAMP = 3;
     // Correct simple test.
     {
         std::vector<memref_t> memrefs = {
-            gen_instr(1, 1), gen_branch(1, 2),
-            gen_instr(1, 3), gen_marker(2, TRACE_MARKER_TYPE_TIMESTAMP, 0),
-            gen_instr(2, 1),
+            gen_instr(TID, 1),     gen_branch(TID, 2),
+            gen_instr(TID, 3),     gen_marker(TID + 1, TRACE_MARKER_TYPE_TIMESTAMP, 0),
+            gen_instr(TID + 1, 1),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Correctly skip CPU ID marker.
```

Comment:
Let's have a case where the branch is taken across the CPU_ID too to ensure the decoding path works there.

Reply:
I've updated the PR to reflect the new approach. PTAL.